#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include <QMovie>
#include <QRegularExpressionValidator>

QT_BEGIN_NAMESPACE
namespace Ui { class MainWindow; }
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();

private:
    Ui::MainWindow *ui;
    QMovie *gifMovie;
    QRegularExpressionValidator *validator;

    QRegularExpression rxCreditCard;
    QRegularExpression rxDate;
    QRegularExpression rxSecCode;

    void loadGif();

    void setValidators();

private slots:
    void btnSendPressed();
};
#endif // MAINWINDOW_H
